torch/csrc/autograd/profiler_utils.cpp

```diff
@@ -0,0 +1,96 @@
+#include <torch/csrc/autograd/profiler_utils.h>
+
+namespace torch { namespace autograd { namespace profiler {
+
+const size_t CONV2D_STRIDE = 3;
+const size_t CONV2D_PADDING = 4;
+const size_t CONV2D_DILATION = 5;
+const size_t CONV2D_GROUPS = 6;
+
+void saveExtraArgs(std::unordered_map<std::string, c10::IValue> &extra_args, const at::RecordFunction& fn) {
+    // switch the type of fn, if convolution or gemm, fill-in extra_args_, then return
+    std::vector<c10::IValue> inputs = fn.inputs();
+
+    TORCH_CHECK(
+      inputs.size() != 0,
+      "Tried to save extra arguments for flops computation, but input shape is not available!");
+
+    if(fn.name() == at::StringView("aten::conv2d")) {
+      at::Tensor input = inputs[0].toTensor();
+      at::Tensor weight = inputs[1].toTensor();
+      extra_args["input_size"] = at::IValue(input.sizes());
+      extra_args["weight_size"] = at::IValue(weight.sizes());
+      extra_args["stride"] = inputs[CONV2D_STRIDE];
+      extra_args["padding"] = inputs[CONV2D_PADDING];
+      extra_args["dilation"] = inputs[CONV2D_DILATION];
+      extra_args["groups"] = inputs[CONV2D_GROUPS];
+    } else if (fn.name() == at::StringView("aten::mm")) {
+      at::Tensor left = inputs[0].toTensor();
+      at::Tensor right = inputs[1].toTensor();
+      extra_args["mat1_size"] = at::IValue(left.sizes());
+      extra_args["mat2_size"] = at::IValue(right.sizes());
+    } else if (fn.name() == at::StringView("aten::mul.Tensor")) {
+      at::Tensor mat = inputs[0].toTensor();
+      extra_args["mat_size"] = at::IValue(mat.sizes());
+    } else if (fn.name() == at::StringView("aten::add.Tensor")) {
+      at::Tensor mat = inputs[0].toTensor();
+      extra_args["mat_size"] = at::IValue(mat.sizes());
+    }
+}
+
+uint64_t computeFlops(const at::StringView &op_name, const std::unordered_map<std::string, c10::IValue> &extra_args) {
```

Comment:
we need to add a C++ unit test that would compute flops by calling this function directly with different argument